docs/principles-for-package-repository-security.md

```diff
@@ -0,0 +1,105 @@
+- Level 3
+  - The package repository supports passwordless authentication, such as passkeys.
+  - The package repository requires MFA for all maintainers.
+  - The package repository requires phishing-resistant MFA for packages deemed critical (e.g. top 1% of packages by downloads). If possible, the package repository supplies security keys to these maintainers.
+
+### Authorization
+
+Applies to: package repositories that have user accounts and accept built packages.
+
+- Level 1
+  - The package repository allows maintainers to provision API keys scoped to specific packages. This allows maintainers to maintain packages via automated workflows without needing to provide their account password.
+- Level 2
+  - The package repository supports role-based access control (RBAC) for maintainers, allowing separate roles for managing users and publishing packages.
+- Level 3
+  - The package repository provisions short-lived identity tokens via  OpenID Connect (OIDC) to prevent needing to provision long-lived API keys. Example: [trusted publishers](https://docs.pypi.org/trusted-publishers/)
+  - The package repository API tokens are integrated into common third-party secret scanning programs like GitGuardian, GitHub, etc. This typically requires using a known prefix or pattern to reduce false-positives.
+  - The package repository supports providing [build provenance](https://repos.openssf.org/build-provenance-for-all-package-registries) for packages.
+
+### General Capabilities
+
+Applies to: all package repositories
+
+- Level 1
+  - The package repository has published a vulnerability disclosure policy allowing security researchers to report vulnerabilities to the package repository, and receive legal safe harbor when doing so. The package repository should strive to remediate reported vulnerabilities within 90 days.
+  - The package repository takes steps to prevent typosquatting attacks around package names. This may include namespacing, detecting similar names, or other actions.
+- Level 2
+  - The package repository has an unpublish policy to prevent specific versions of a package from being replaced, and, if a package is deleted, prevent others from reusing the same package name and version.
+  - The package repository allows users to report a package as suspicious or malicious via the registry UI, tooling CLI, and via API.
+  - The package repository takes steps to detect malware, including scanning for known malware code snippets and hashes.
+- Level 3
+  - Undergo periodic cloud security review (and optionally publish remediations once they are complete)
+  - The package repository publishes an event transparency log to enable detection of anomalous behaviors (e.g. [replicate.npmjs.com](https://replicate.npmjs.com/)).
+  - Malicious packages are published using OSV (e.g. [here](https://github.com/ossf/malicious-packages))
+
+### CLI Tooling
```

Comment:
This seems overly broad, are we really just considering specific classes of CLI tooling, e.g. "installers", "builders", "publishers", etc?

Reply:
Agreed -- if the doc is meant to be taxonomical, it might make sense to clarify that tooling here is meant to be the "package installer" (based on context).